### PR TITLE
Add sentry replay to config

### PR DESCRIPTION
**Pull request by adonispuente**
In an effort to make sentry errors more understandable (per my squad assignment), I want to add the replay functionality to our config. This allows us to watch the replays similar to hotjar, except with sentry now we'll know what error is associated to what bug. 

I've added the sentry-mask class to block certain aspects that hotjar was also blocking. I also added sentrys capture message function to report errors above 400 as a warning as opposed to an error

I followed the instructions found here 
https://docs.sentry.io/platforms/javascript/guides/react/session-replay/?_ga=2.107964160.348271133.1675046398-87740463.1664195656

### CODE REVIEW

**Review by Hyperkid123 — APPROVED** [review]

**Comment by Hyperkid123** [comment]
Had to re-open the PR to wake up the bot

File: src/components/FeatureFlags/FeatureFlagsProvider.tsx [hunk 1]
```diff
@@ -22,8 +23,9 @@ const config: IFlagProvider['config'] = {
       .fetch(url, headers)
       .then((resp) => {
         // prevent the request from falling back to default error behavior
```
Comment (jiridostal) [inline src/components/FeatureFlags/FeatureFlagsProvider.tsx:25]: Correct me if I'm wrong, but doesn't this change "re-enable" bubbling this error up to the default error handler?

Comment (Hyperkid123) [reply src/components/FeatureFlags/FeatureFlagsProvider.tsx:25]: No, there is a catch branch below, that is the actual error handler.

Comment (jiridostal) [reply src/components/FeatureFlags/FeatureFlagsProvider.tsx:25]: I see, thanks!

**Pull request merged**